### PR TITLE
[MATRIX-191] only pass secret for create and alter model

### PR DESCRIPTION
**Pull request by lihaosky**
Release Notes
-------------
bug fix:
   [early access] only pass secrets for create and alter model request

Breaking Changes
- PLACEHOLDER

New Features
- PLACEHOLDER

Bug Fixes
- PLACEHOLDER

Checklist
---------
- [ ] Leave this box unchecked if features are not yet available in production

What
----

References
----------

Test & Review
-------------

### CODE REVIEW

File: pkg/flink/internal/store/store_utils.go [hunk 1]
```diff
@@ -392,6 +392,39 @@ func useError() *types.StatementError {
 	}
 }
 
+func maybeCreateAlterModelStatement(statement string) bool {
+	regex := regexp.MustCompile(config.CreateAlterModelRegex)
+	matches := regex.FindAllStringSubmatch(statement, -1)
+	names := regex.SubexpNames()
+	fmt.Println(names)
+	//return match != nil && len(match) == 2 && len(match[1]) != 0
+	fmt.Println(matches)
```
Comment (brianstrauch) [inline pkg/flink/internal/store/store_utils.go:401]: Please don't commit commented code or `Println()` functions

File: pkg/flink/config/local_statements.go [hunk 2]
```diff
@@ -22,6 +22,27 @@ const (
...
+	/*
+	   Create Alter model regex. Match create|alter model with
+	     1. allow any space,return,tab between them
+	     2. They are not surrounded by unescaped single quote
+	     3. Single quote is escaped by doubling them like ''.
+	     4. Start beginning of line, with space or `;`
+
+	     (?msi) -> enable multiline, . match \n and case insensitive
+	     \s -> matches space, \n, \t, \r, \f
+	     '' -> match escaped single quote
```
Comment (chmeyers) [inline pkg/flink/config/local_statements.go:35]: Do we have to make sure it's not inside a comment as well?

File: pkg/flink/internal/store/store.go [hunk 3]
```diff
@@ -105,6 +105,9 @@ func (s *Store) ProcessStatement(statement string) (*types.ProcessedStatement, *
 }
 
 func createSqlV1Statement(statement string, statementName string, computePoolId string, properties map[string]string) flinkgatewayv1.SqlV1Statement {
+	if !maybeCreateAlterModelStatement(statement) {
```
Comment (chmeyers) [inline pkg/flink/internal/store/store.go:108]: Would it make sense to only send those secret properties that are actually referenced by the SQL, or does ALTER MODEL need them even if they aren't listed?

Comment (lihaosky) [reply pkg/flink/internal/store/store.go:108]: alter model can also reference it. Make sense to check they are used

**Review by Minggedeng — APPROVED** [review]

**Comment by lihaosky** [comment]
Don't need this anymore. Will fix in gateway

**Pull request closed without merging**